Sources/MapboxMaps/Camera/CameraOptionsInterpolator.swift

```diff
@@ -0,0 +1,68 @@
+import CoreLocation
+import UIKit
+
+internal protocol CameraOptionsInterpolatorProtocol: AnyObject {
+    func interpolate(from: CameraOptions,
+                     to: CameraOptions,
+                     fraction: Double) -> CameraOptions
+}
+
+internal final class CameraOptionsInterpolator: CameraOptionsInterpolatorProtocol {
+    private let optionalInterpolator = OptionalInterpolator()
+    private let coordinateInterpolator: CoordinateInterpolatorProtocol
+    private let uiEdgeInsetsInterpolator: UIEdgeInsetsInterpolatorProtocol
+    private let interpolator: InterpolatorProtocol
+    private let bearingInterpolator: InterpolatorProtocol
+
+    internal init(coordinateInterpolator: CoordinateInterpolatorProtocol,
+                  uiEdgeInsetsInterpolator: UIEdgeInsetsInterpolatorProtocol,
+                  interpolator: InterpolatorProtocol,
```

Comment:
Should this parameter name indicate what kind of interpolator is expected? It seems it is used to interpolate values on two different scales(zoom 0-22 and pitch 0-85).

Reply:
Most of this comment is addressed in my replies and changes below, but one additional note here is that zoom and pitch can use the same interpolator since the interpolation works the same way for both (interpolation of double values without any special wrapping or other logic).